src/benchmarks/gc/src/analysis/process_trace.py

```diff
@@ -37,39 +43,36 @@
 from .mechanisms import get_mechanisms_and_reasons_for_process_info
 from .types import (
     Failable,
+    get_gc_kind_for_abstract_trace_gc,
     MaybeMetricValuesForSingleIteration,
     ProcessedGC,
     ProcessedHeap,
     ProcessInfo,
     ProcessedTrace,
-    ProcessQuery,
     RunMetrics,
     ThreadToProcessToName,
 )
 
 
-def test_result_from_path(path: Path) -> TestResult:
+def test_result_from_path(path: Path, process: ProcessQuery) -> TestResult:
     if path.name.endswith(".yaml"):
         # Try to find trace path
+        assert (
+            process is None
+        ), f"When working with .yaml files, --process is not needed. Check {path}."
         trace_file_name = load_test_status(path).trace_file_name
         return TestResult(
             test_status_path=path, trace_path=map_option(trace_file_name, lambda n: path.parent / n)
         )
     else:
         assert is_trace_path(path), f"{path} should be a '.yaml' test output file or a trace file."
-        # User might have specified the tracefile when a status file still exists
-        test_status = change_extension(path, ".yaml")
-        status_exists = test_status.exists()
-        assert (not status_exists) or load_test_status(test_status).trace_file_name == path.name
-        return TestResult(test_status_path=test_status if status_exists else None, trace_path=path)
+        return TestResult(
+            test_status_path=None, trace_path=path, process=_convert_to_tuple(process)
+        )
 
 
 def get_processed_trace(
-    clr: Clr,
-    test_result: TestResult,
-    process: ProcessQuery,
-    need_mechanisms_and_reasons: bool,
-    need_join_info: bool,
+    clr: Clr, test_result: TestResult, need_mechanisms_and_reasons: bool, need_join_info: bool
```

Comment:
I assume this was done by the linter. Parameter list got shorter and was reformatted.